### PR TITLE
Fetch serial no by default

### LINKED ISSUE #6645: Serial Number by default FIFO for Production Order

### PR DESCRIPTION
**Pull request by KanchanChauhan**
Fixed https://github.com/frappe/erpnext/issues/6645

### CODE REVIEW

**Comment by bcornwellmott** [comment]
I'm concerned about having this happen automatically without any user interaction. Users could forget this is a serial numbered part and transfer incorrect serial numbers without any warning or prompt. Could we have this trigger on a button press or a checbox in the Stock Settings?

**Comment by KanchanChauhan** [comment]
@bcornwellmott It will happen only if "Automatically Set Serial Nos based on FIFO" is checked in Stock Settings.

File: erpnext/stock/doctype/stock_entry/stock_entry.js [hunk 1]
```diff
@@ -409,6 +430,15 @@ cur_frm.cscript.item_code = function(doc, cdt, cdn) {
 	}
 }
 
+
+cur_frm.cscript.qty = function(doc, cdt, cdn) {
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry/stock_entry.js]: Always use `frappe.ui.form.on`

File: erpnext/stock/doctype/stock_entry/stock_entry.js [hunk 2]
```diff
@@ -305,6 +305,25 @@ erpnext.stock.StockEntry = erpnext.stock.StockController.extend({
 
 		this.frm.set_value("total_additional_costs", flt(total_additional_costs, precision("total_additional_costs")));
 	},
+
+	set_serial_no: function(doc, cdt, cdn) {
+		var d = locals[cdt][cdn];
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry/stock_entry.js]: return if no `s_warehouse && item_code && qty`

File: erpnext/stock/doctype/stock_entry/stock_entry.py [hunk 3]
```diff
@@ -493,6 +494,9 @@ def get_item_details(self, args=None, for_update=False):
 		stock_and_rate = args.get('warehouse') and get_warehouse_details(args) or {}
 		ret.update(stock_and_rate)
 
+		if args.get('fetch_serial_no'):
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: Actually it does not make any sense, as qty is always zero.

File: erpnext/stock/get_item_details.py [hunk 4]
```diff
@@ -339,7 +339,7 @@ def get_pos_profile(company):
 	return pos_profile and pos_profile[0] or None
 
 
-def get_serial_nos_by_fifo(args, item_doc):
+def get_serial_nos_by_fifo(args, item_doc = None):
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: `item_doc` is not used

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/stock/get_item_details.py (hunk 4 above)
Comment (KanchanChauhan) [inline erpnext/stock/get_item_details.py]: All changes done

**Review by nabinhait — APPROVED** [review]

**Pull request merged**